ts/main.ts

```diff
@@ -1,64 +1,92 @@
   };
-  [key: string]: any; // To account for any additional fields
+  rating: string;
 }
 
 const animeContainer = document.querySelector(
   '.anime-container',
 ) as HTMLDivElement;
 
-async function fetchAnime(): Promise<Anime[]> {
+async function fetchAnime(): Promise<void> {
   try {
     const response = await fetch('https://api.jikan.moe/v4/top/anime');
+    if (!response.ok) throw new Error('Fetch for top anime failed!');
     const data = await response.json();
-    return data.data as Anime[];
+
+    data.data.forEach((anime: Anime) => {
+      animeContainer.appendChild(renderAnime(anime));
+    });
   } catch (error) {
     console.error('Error fetching anime data:', error);
-    return [];
   }
 }
 
-function renderAnime(animeList: Anime[]): void {
-  animeContainer.innerHTML = '';
+// REnders the anime cards
+function renderAnime(anime: Anime): HTMLDivElement {
+  const animeDiv = document.createElement('div');
+  animeDiv.classList.add('animes');
 
-  animeList.forEach((anime) => {
-    const animeDiv = document.createElement('div');
-    animeDiv.classList.add('animes');
+  const imageDiv = document.createElement('div');
+  imageDiv.classList.add('image-container');
+  const img = document.createElement('img');
+  img.src = anime.images.jpg.image_url;
+  img.alt = anime.title;
+  imageDiv.appendChild(img);
 
-    const imageDiv = document.createElement('div');
-    imageDiv.classList.add('image-container');
-    const img = document.createElement('img');
-    img.src = anime.images.jpg.image_url;
-    img.alt = anime.title;
-    imageDiv.appendChild(img);
+  const textDiv = document.createElement('div');
+  textDiv.classList.add('text-container');
+  const title = document.createElement('h2');
+  title.textContent = anime.title;
+  const episodes = document.createElement('p');
+  episodes.textContent = `Episodes: ${anime.episodes ?? 'N/A'}`;
 
-    const textDiv = document.createElement('div');
-    textDiv.classList.add('text-container');
-    const title = document.createElement('h2');
-    title.textContent = anime.title;
-    const episodes = document.createElement('p');
-    episodes.textContent = `Episodes: ${anime.episodes ?? 'N/A'}`;
+  textDiv.appendChild(title);
+  textDiv.appendChild(episodes);
 
-    textDiv.appendChild(title);
-    textDiv.appendChild(episodes);
+  // Add an event listener to the title to swap views
+  title.addEventListener('click', () => {
+    swapView(anime);
+  });
 
-    animeDiv.appendChild(imageDiv);
-    animeDiv.appendChild(textDiv);
+  animeDiv.appendChild(imageDiv);
+  animeDiv.appendChild(textDiv);
 
-    animeContainer.appendChild(animeDiv);
-  });
+  return animeDiv;
 }
 
-async function init(): Promise<void> {
-  const animeList = await fetchAnime();
-  renderAnime(animeList);
+// SWAPS THE VIEW . im pretty sure idk it wasnt working for some reason but now it is
+function swapView(anime: Anime): void {
+  const detailView = document.querySelector('.detail-view') as HTMLDivElement;
```

Comment:
I would be wary of the comments you leave since these will be visible. If you want to leave comment you can leave them about _why_ the code is that rather than _what_ it is doing if possible. Otherwise you could leave some in for organization but you want the comments to be professional/good looking.